mlibrary/lib.umich.edu
# accessibility updates to staff-directory table

# Issue Overview
This is a violation in the [baseline accessibility evaluation](https://docs.google.com/document/d/1PGCPqSdbGEgx83_zMgkJvoCHYd3dhtQleoZBwPAgKzQ/edit?usp=sharing).

On the [Staff Directory](https://lib.umich.edu/about-us/staff-directory?): (1) the caption of the table is always “420 results”. This is announced regardless of how many results are showing which is confusing for the screen reader user, and (2) there is an empty role and an invalid attribute on the caption (probably meant to be a role=”status” live region.). This is likely also an issue on [Find a Specialist](https://lib.umich.edu/research-and-scholarship/help-research/find-specialist).

# Fixes Overview
To address the issues above, this PR fixes the following:
- Has a visible caption below the search and above the results.
- Adds a `aria-live='polite'` to the search results caption (not the results themselves, but the caption).
- BONUS! I also took this time to add the `scope` attribute to the table columns and rows.

Additional styling choices were made for the caption. The filter variables are bolded, the caption is left-aligned, and the background and underline from the `<alert>` component are removed.

I'm no longer using the "alert" component since the alert is primarily providing styles (which we no longer want) and an empty `role` attribute.

This was also identified as a potential enhancement in post-launch UX research: WEBSITE-23: Explore adding search summary to Staff Directory and Find Specialist

> This pull request fixes [WEBSITE-137 Remediate violations on dynamic tables](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-137)

> And [WEBSITE-23 Explore adding search summary to Staff Directory and Find Specialist](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-23)

## Notes
The screen reader `aria-live="polite"` may be delayed when reading the results on the table. I could implement a delay on readback but I need to read up on the best course of action for this before I commit to a solution.

## Testing
To test this, use a screen reader to access the STAFF PAGE PREVIEW URL and the FIND A SPECIALIST PREVIEW URL use the search input to ensure it's reading the results properly. The results should NOT be updating after every input as to not annoyingly interrupt the user UNLESS there are no results.

- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Use a screen reader to test the following pages:
  - [ ] [STAFF PAGE PREVIEW URL](https://deploy-preview-300--future-wwwlib-previews.netlify.app/about-us/staff-directory?)
  - [ ] [FIND A SPECIALIST PREVIEW URL](https://deploy-preview-300--future-wwwlib-previews.netlify.app/research-and-scholarship/help-research/find-specialist)
